File: 2015/1/part-two.py
#!/usr/bin/env python3

with open('input', 'r+') as f:
    content = f.read()
    floor = 0

    for idx, char in enumerate(content):
        floor = floor + (1 if char == '(' else -1)

        if floor < 0:
            print(idx)
            exit()

File: 2015/2/part-two.rb
#!/usr/bin/env ruby

# "1x1x10"

puts File.read("input-2").lines
  .map(&:strip)
  .map { |s| s.split("x").map(&:to_i) }
  .map { |l, w, h|
    a, b = [l, w, h].min(2)

    a + a + b + b + (l * w * h)
  }
  .sum




